DS/DsAssignment4/PrintApplication.java

```diff
@@ -0,0 +1,57 @@
+public class PrintApplication {
+
+	private PriorityQueue priorityQueue = new PriorityQueue(20);
+	public boolean printerStatus;
+
+	private boolean isPrinterFree() {
+		return printerStatus;
+
+	}
+
+	public String processRequest(Task task) {
+		String result = "";
+		if (isPrinterFree() && priorityQueue.checkIsEmpty()) {
+			result = print(task);
+
+		} else {
+			priorityQueue.insert(task);
+			System.out.println("you are in queue...plz wait for some time");
+			result = "you are in queue...plz wait for some time";
+		}
+
+		return result;
+
+	}
+
+	public String processQueue() {
+		String result = "";
+		while (!priorityQueue.checkIsEmpty()) {
+			if (isPrinterFree()) {
+
+				Task task = priorityQueue.remove();
+				result = result + print(task) + "/";
+
+			}
+
+		}
+		return result;
+
+	}
+
+	private String print(Task task) {
+		String requestHolder = "";
+		if (task.priority == 4)
```

Comment:
ACTION: Why are you checking priorities in the "print"? What is the use of priority-queue then? You print method should only use pop or "find-min" method (of the heap). The priority should be only be considered by the "insert" method of the queue/heap.